src/components/multi-action-button/multi-action-button.component.tsx

```diff
@@ -104,38 +104,59 @@ export const MultiActionButton = ({
       );
       let nextIndex = -1;
 
-      if (Events.isUpKey(ev)) {
-        nextIndex =
-          currentIndex > 0 ? currentIndex - 1 : buttonChildren.length - 1;
+      const refocusMainButton = () => {
+        setShowAdditionalButtons(false);
+        buttonRef.current?.focus();
         ev.preventDefault();
-      }
+      };
 
-      if (Events.isDownKey(ev)) {
-        nextIndex =
-          currentIndex < buttonChildren.length - 1 ? currentIndex + 1 : 0;
-        ev.preventDefault();
+      if (Events.isUpKey(ev) && currentIndex > 0) {
+        nextIndex = currentIndex - 1;
       }
 
-      if (Events.isTabKey(ev)) {
-        const elements = Array.from(
-          document.querySelectorAll(
-            defaultFocusableSelectors
-          ) as NodeListOf<HTMLElement>
-        ).filter((el) => Number(el.tabIndex) !== -1);
-
-        const indexOf = elements.indexOf(
-          buttonRef.current as HTMLButtonElement
-        );
-
-        elements[indexOf]?.focus();
+      if (Events.isDownKey(ev) && currentIndex < buttonChildren.length - 1) {
+        nextIndex = currentIndex + 1;
+      }
 
-        // timeout enforces that the "hideButtons" method will be run after browser focuses on the next element
-        setTimeout(hideButtons, 0);
+      const tabPressed = Events.isTabKey(ev);
+      const tabShiftPressed = tabPressed && Events.isShiftKey(ev);
+
+      if (tabShiftPressed) {
+        if (currentIndex === 0) {
+          refocusMainButton();
+        } else {
+          nextIndex = currentIndex - 1;
+          ev.preventDefault();
+        }
+      } else if (tabPressed) {
+        if (currentIndex === buttonChildren.length - 1) {
+          const elements = Array.from(
+            document.querySelectorAll(
+              defaultFocusableSelectors
+            ) as NodeListOf<HTMLElement>
+          ).filter((el) => Number(el.tabIndex) !== -1);
+
+          const indexOf = elements.indexOf(
+            buttonRef.current as HTMLButtonElement
+          );
+
+          elements[indexOf]?.focus();
```

Comment:
I realise this wasn't your code and existed before, but is all this really necessary? We're finding the index of buttonRef.current in the elements array, then grabbing the element at that index, and focusing it. In other words, we're focusing the main button which should be accessible at buttonRef.current.

At the risk of sounding stupid, why can't we do this with just buttonRef.current?.focus ?? (I'm not even sure the ?. is needed there but it does no harm and prevents crashes in potential weird situations.)

Reply:
It seems that according to the new requirements it should be `elements[indexOf + 1]?.focus();`